cypress/tests/index.spec.ts

```diff
@@ -1,10 +1,92 @@
 /// <reference types="cypress" />
-describe("example to-do app", () => {
+
+const DEFAULT_URL = Cypress.env('CYPRESS_BASE_URL');
+const INVALID_PRIVATE_KEY = 'GGAJOGAJOPAGJOPGAJOPGAJPAGPJOA08421';
+const VALID_PRIVATE_KEY = 'SAZ5WSF7U7KB43CAU4IU7TMYNLH67S3WN4I42GHLGHMOCJBC7CPLDHTV';
+const INVALID_KEY_MESSAGE = 'Invalid secret key';
+
+context("Index", () => {
   beforeEach(() => {
-    cy.visit("http://localhost:3000");
+    cy.visit(DEFAULT_URL);
+    cy.get('[data-cy="keys-button"]').as('getKeysButton');
+    cy.get('[data-cy="connect-button"]').as('connectButton');
+  });
+
+  describe('Index page', () => {
+    it('Should exist "Generate Key" button', () => {
+      cy.get('@getKeysButton').should('exist');
+    });
+
+    it('Should exist "Connect with secret key" button', () => {
+      cy.get('@connectButton').should('exist');
+    })
+  });
+
+  describe('Key Modal', () => {
+    it('Should open modal when "Generate Key" button is clicked', () => {
+      cy.get('@getKeysButton').click();
+      cy.get('[data-cy="key-modal"]').should('be.visible');
+    });
+
+    it('Should provide a pair of keys', () => {
+      cy.get('@getKeysButton').click();
+      cy.get('[data-cy="private-key"]').invoke('text')
+        .should('not.be.empty')
+        .should('have.length.at.least', 50);
+
+      cy.get('[data-cy="public-key"]').invoke('text')
+        .should('not.be.empty')
+        .should('have.length.at.least', 50);
+    });
+
+    it('Should copy keys when copy button is clicked', () => {
+      cy.get('@getKeysButton').click();
+      cy.get('[data-cy="copy-button"]').click();
+
+      cy.window().its('navigator.clipboard').invoke('readText')
+        .should('have.length.above',100)
```

Comment:
Is Prettier not working well?
```suggestion
        .should('have.length.above', 100)
```

Reply:
Sorry, i forgot to run our formatter.